### ISSUE TITLE
org.eolang.cage must prohibit different object "types"

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#1307]
The `cage` object must return an `error` if we try to `.write` an object that is not a copy of the abstract object that was saved before.

**Comment by yegor256** [comment]
@Graur please, help

**Comment by Graur** [comment]
@yegor256 Do we need to save the type of the memoized object? Otherwise we can check its length

**Comment by yegor256** [comment]
@maxonfjvipon please, help

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look at this.
```
[x] > obj
  x > @
obj 1 > obj1

# Case 1
cage obj > cg
cg.write obj1

# Case 2
cage obj1 > cg
cg.write obj
```

1. Init `cage` with abstract object `obj` then write its copy `obj1` with bounded `x`
2. Init `cage` with copy (of `obj`) `obj1` then write abstract `obj`

Should these both cases be legal?

**Comment by yegor256** [comment]
@maxonfjvipon both cases should be illegal, in my opinion

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#2335](https://github.com/objectionary/eo/issues/2335), [#2336](https://github.com/objectionary/eo/issues/2336) are still not solved.

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#2335](https://github.com/objectionary/eo/issues/2335), [#2373](https://github.com/objectionary/eo/issues/2373) are still not solved; solved: [#2336](https://github.com/objectionary/eo/issues/2336).

**Comment by 0pdd** [comment]
@yegor256 all 3 puzzles are solved here: [#2335](https://github.com/objectionary/eo/issues/2335), [#2336](https://github.com/objectionary/eo/issues/2336), [#2373](https://github.com/objectionary/eo/issues/2373).
